File: data/multilang_comparison/program_julia/src/func25.c
#include "program_julia.h" 
array_t* func25(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop31 = 0;
   unsigned int loopLimit31 = (rand()%loopsFactor)/3 + 1;
   for(; loop31 < loopLimit31; loop31++) {
      if(PATH0 & 1) {
         array_t* array100;
         if (pCounter > 0) {
            array100 = vars->data[--pCounter];
            array100->refC++;
            DEBUG_COPY(array100->id);
         } else {
            array100 = (array_t*)malloc(sizeof(array_t));
            array100->size = 127;
            array100->refC = 1;
            array100->id = 100;
            array100->data = (unsigned int*)malloc(array100->size*sizeof(unsigned int));
            memset(array100->data, 0, array100->size*sizeof(unsigned int));
            DEBUG_NEW(array100->id);
         }
         unsigned int loop32 = 0;
         unsigned int loopLimit32 = (rand()%loopsFactor)/4 + 1;
         for(; loop32 < loopLimit32; loop32++) {
            for (int i = 0; i < array100->size; i++) {
               array100->data[i]--;
            }
            for (int i = 0; i < array100->size; i++) {
               if (array100->data[i] == 58) { 
                  return array100;
               }
            }
         }
         array_t_param params0;
         params0.size = 1;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array100;
         array_t* array101 = func42(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array101->id);
         free(params0.data);
         array101->refC--;
         if(array101->refC == 0) {
            free(array101->data);
            free(array101);
            DEBUG_FREE(array101->id);
         }
         array100->refC--;
         if(array100->refC == 0) {
            free(array100->data);
            free(array100);
            DEBUG_FREE(array100->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 0;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         array_t* array102 = func34(&params0, get_path(), loopsFactor);
         DEBUG_RETURN(array102->id);
         free(params0.data);
         array102->refC--;
         if(array102->refC == 0) {
            free(array102->data);
            free(array102);
            DEBUG_FREE(array102->id);
         }
      }
      array_t* array103;
      if (pCounter > 0) {
         array103 = vars->data[--pCounter];
         array103->refC++;
         DEBUG_COPY(array103->id);
      } else {
         array103 = (array_t*)malloc(sizeof(array_t));
         array103->size = 396;
         array103->refC = 1;
         array103->id = 103;
         array103->data = (unsigned int*)malloc(array103->size*sizeof(unsigned int));
         memset(array103->data, 0, array103->size*sizeof(unsigned int));
         DEBUG_NEW(array103->id);
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array103;
      array_t* array104 = func35(&params0, loopsFactor);
      DEBUG_RETURN(array104->id);
      free(params0.data);
      if(PATH0 & 2) {
         array_t* array105;
         if (pCounter > 0) {
            array105 = vars->data[--pCounter];
            array105->refC++;
            DEBUG_COPY(array105->id);
         } else {
            array105 = (array_t*)malloc(sizeof(array_t));
            array105->size = 84;
            array105->refC = 1;
            array105->id = 105;
            array105->data = (unsigned int*)malloc(array105->size*sizeof(unsigned int));
            memset(array105->data, 0, array105->size*sizeof(unsigned int));
            DEBUG_NEW(array105->id);
         }
         unsigned int loop33 = 0;
         unsigned int loopLimit33 = (rand()%loopsFactor)/4 + 1;
         for(; loop33 < loopLimit33; loop33++) {
            for (int i = 0; i < array104->size; i++) {
               array104->data[i]--;
            }
            for (int i = 0; i < array103->size; i++) {
               if (array103->data[i] == 72) { 
                  return array103;
               }
            }
         }
         array_t_param params1;
         params1.size = 3;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array103;
         params1.data[1] = array104;
         params1.data[2] = array105;
         array_t* array106 = func46(&params1, get_path(), loopsFactor);
         DEBUG_RETURN(array106->id);
         free(params1.data);
         array106->refC--;
         if(array106->refC == 0) {
            free(array106->data);
            free(array106);
            DEBUG_FREE(array106->id);
         }
         array105->refC--;
         if(array105->refC == 0) {
            free(array105->data);
            free(array105);
            DEBUG_FREE(array105->id);
         }
      }
      else {
         array_t_param params1;
         params1.size = 2;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array103;
         params1.data[1] = array104;
         array_t* array107 = func40(&params1, loopsFactor);
         DEBUG_RETURN(array107->id);
         free(params1.data);
         array107->refC--;
         if(array107->refC == 0) {
            free(array107->data);
            free(array107);
            DEBUG_FREE(array107->id);
         }
      }
      for (int i = 0; i < array104->size; i++) {
         if (array104->data[i] == 40) { 
            return array104;
         }
      }
      array104->refC--;
      if(array104->refC == 0) {
         free(array104->data);
         free(array104);
         DEBUG_FREE(array104->id);
      }
      array103->refC--;
      if(array103->refC == 0) {
         free(array103->data);
         free(array103);
         DEBUG_FREE(array103->id);
      }
   }
   array_t* array108;
   if (pCounter > 0) {
      array108 = vars->data[--pCounter];
      array108->refC++;
      DEBUG_COPY(array108->id);
   } else {
      array108 = (array_t*)malloc(sizeof(array_t));
      array108->size = 42;
      array108->refC = 1;
      array108->id = 108;
      array108->data = (unsigned int*)malloc(array108->size*sizeof(unsigned int));
      memset(array108->data, 0, array108->size*sizeof(unsigned int));
      DEBUG_NEW(array108->id);
   }
   return array108;
}

File: data/multilang_comparison/program_julia/src/func25.jl
include("program_julia_head.jl")
function func25(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   loopLimit22 = (50)/3 + 1
   for i in 1:loopLimit22 
      if (PATH0 & 1) != 0
         array69 = nothing
         if pCounter > 1
             array69 = vars.data[pCounter]
             pCounter -= 1
             array69.refC += 1
         else
             array69 = JArray(zeros(Int, 828), 828, 1, 69)
         end
         loopLimit23 = (50)/4 + 1
         for i in 1:loopLimit23 
            for i in 0:(array69.size-1) 
               array69.data[i+1] -= 1
            end
            for i in 1:array69.size
                if array69.data[i] == 46
                        return array69
               end
            end
         end
         params0 = JArray_param{Int}()
         params0.size = 1
         push!(params0.data, array69)
         params0.size += 1
         array70 = func42(params0, get_path(), loopsFactor);
         array70.refC -= 1
         if array70.refC == 0
             empty!(array70.data)
             array70 = nothing
         end
         array69.refC -= 1
         if array69.refC == 0
             empty!(array69.data)
             array69 = nothing
         end
      
      else 
         params0 = JArray_param{Int}()
         params0.size = 0
         array71 = func34(params0, loopsFactor);
         array71.refC -= 1
         if array71.refC == 0
             empty!(array71.data)
             array71 = nothing
         end
      end
      array72 = nothing
      if pCounter > 1
          array72 = vars.data[pCounter]
          pCounter -= 1
          array72.refC += 1
      else
          array72 = JArray(zeros(Int, 810), 810, 1, 72)
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array72)
      params0.size += 1
      array73 = func35(params0, loopsFactor);
      if (PATH0 & 2) != 0
         array74 = nothing
         if pCounter > 1
             array74 = vars.data[pCounter]
             pCounter -= 1
             array74.refC += 1
         else
             array74 = JArray(zeros(Int, 661), 661, 1, 74)
         end
         loopLimit24 = (50)/4 + 1
         for i in 1:loopLimit24 
            for i in 0:(array72.size-1) 
               array72.data[i+1] -= 1
            end
            for i in 1:array73.size
                if array73.data[i] == 5
                        return array73
               end
            end
         end
         array74.refC -= 1
         if array74.refC == 0
             empty!(array74.data)
             array74 = nothing
         end
      
      else 
         params1 = JArray_param{Int}()
         params1.size = 2
         push!(params1.data, array72)
         params1.size += 1
         push!(params1.data, array73)
         params1.size += 1
         array75 = func40(params1, loopsFactor);
         array75.refC -= 1
         if array75.refC == 0
             empty!(array75.data)
             array75 = nothing
         end
      end
      for i in 1:array72.size
          if array72.data[i] == 26
                  return array72
         end
      end
      array73.refC -= 1
      if array73.refC == 0
          empty!(array73.data)
          array73 = nothing
      end
      array72.refC -= 1
      if array72.refC == 0
          empty!(array72.data)
          array72 = nothing
      end
   end
   array76 = nothing
   if pCounter > 1
       array76 = vars.data[pCounter]
       pCounter -= 1
       array76.refC += 1
   else
       array76 = JArray(zeros(Int, 522), 522, 1, 76)
   end
   return array76;
end

